### ISSUE TITLE
Brazilian 9 digit number not accepted

### ISSUE THREAD

**Issue by dswebdotnet** [googlei18n/libphonenumber#1217]
In Brazil they have 9 digit mobile phone numbers for some cities and 8 digits for other. LibPhoneNumber doesn't seem to validate 9-digit numbers

For example:

City code 51 is for the city of Porto Alegre, where they have a 9 digit format mobile phone number.

example number:
51986586134

what can be done to rectify this?

Apologies if issue report is not conclusive enough.

Thanks.

**Comment by keghani** [comment]
The transition for 4x and 5x area codes in Brazil is scheduled for November according to http://www.anatel.gov.br/Portal/verificaDocumentos/documento.asp?numeroPublicacao=337603&filtro=1&documentoPath=337603.pdf

Was this schedule updated?

**Comment by dswebdotnet** [comment]
Sorry I do not the answer to this. I'm just aware from a client who's been trying to use 9 digit numbers that they're not validated.

Can we expect LibPhoneNumber to be updated to support this standard and if so, what would the timeline be?

Thanks.

**Comment by keghani** [comment]
We are planning to update the numbers for area codes 41-49 and 51-55 to have the 9th digit by November. This is in line with the official documentation from Anatel referenced above.

Maybe the client has a typo?

To support the renumberings earlier than announced, we'd need evidence that the transitions have already taken place. Please see our [contribution guidelines](https://github.com/googlei18n/libphonenumber/blob/master/CONTRIBUTING.md#provide-supporting-evidence).

**Comment by dswebdotnet** [comment]
Thanks Keghani for your insightful information. It's been really helpful. I have gone back to my client asking for an update on the transition to the 9 digits in Brazil. 

It would be ideal to keep this ticket open until such information can be provided. Then we can following the contribution guidelines to potentially update the database.

Thanks!

**Comment by keghani** [comment]
Thanks @dswebdotnet! Looking forward to hearing back from the client :)

**Comment by keghani** [comment]
@dswebdotnet Would you be able to ask your client whether they can use `51 8658 6134` instead of `51 9 8658 6134`? It is possible that some carriers started renumbering early, but not the entirety of new ranges is valid yet.

This issue is blocked on your or other GitHub contributors' input. Please help us find evidence that the next phase of renumberings in Brazil has taken place ahead of the announced schedule. Otherwise we don't have the required information to fix this and may have to close it as infeasible.

**Comment by keghani** [comment]
I'm closing this issue as we didn't get an early notification.
We'll support all the 4x and 5x area codes with the 9th digit before they are put in use in November, as announced by official sources in Brazil.

**Comment by diogenes** [comment]
Hi @keghani!

We're currently using the `7.5.0` version and I'm not sure if a 9th digit is supported for the 4x area. Do you know what is the first version supporting it?

Thanks!

**Comment by keghani** [comment]
Yes, last week's release, 7.7.3.

**Comment by diogenes** [comment]
Thanks, @keghani!

**Comment by jantonho** [comment]
Hi @keghani In Brazil is required using 9 digits now. It's necessary disregard 8 digits for all areas

**Comment by keghani** [comment]
Thanks @jantonho!

We are aware that some of the old numbers that we're considering valid are no longer connecting.
However, we made a conscious decision to keep considering the ranges valid because some old numbers in those ranges are actually still able to connect, depending on the carrier.

We will eventually stop supporting the old ranges as valid. This is not high priority and we probably won't do it unless there is very strong evidence that *absolutely no numbers* in the old ranges are still working.

**Comment by jantonho** [comment]
After 2/14/2017 calls made using 8 digits are not completed. SMS and MMS are not sent. As defined by Anatel http://www.anatel.gov.br/setorregulado/perguntas-frequentes?view=faq&catid=1

Thanks @keghani

**Comment by keghani** [comment]
@jantonho Thanks for the report! I'm tracking removal at https://github.com/googlei18n/libphonenumber/issues/1618
